cmd/test/test.go

```diff
@@ -224,6 +230,28 @@ func newTestCommand() *cobra.Command {
 	return &cmd
 }
 
+// Special error handling for appstudio format only
+func appstudioErrorHandler(noFail bool, prefix string, err error) error {
+	// Output some json to stdout
+	applicationsnapshot.OutputAppstudioReport(applicationsnapshot.AppstudioReportForError())
+
+	// Beware we're effectively changing the meaning of the --no-fail flag here.
+	// Rather than being only about policy failures any more, we're extending
+	// it to also mean don't return a non-zero exit code for an error handled
+	// by this function.
+	if noFail {
```

Comment:
Is this something we want to replicate in other commands?

Reply:
Hmm, maybe. It's a similar to --strict used elsewhere.